#!/usr/bin/env python
# -*- coding:UTF-8 -*-

PACKAGE = "dynamic_calibration"

from dynamic_reconfigure.parameter_generator_catkin import *
from math import pi as pi

gen = ParameterGenerator() #创建一个参数生成器对象

#gen.add(参数名称，参数类型，位掩码，参数描述，默认值，最小值，最大值)
#type:int_t, double_t, str_t, bool_t
#gen.add(name, type, level, description, default, min, max)
gen.add( "x", double_t, 0, "move in x", 0.0, -5.0, 5.0)
gen.add( "y", double_t, 0, "move in y", 0.0, -5.0, 5.0)
gen.add( "z", double_t, 0, "move in z", 0.0, -5.0, 5.0)
gen.add( "theta_x", double_t, 0, "twist in x", pi/2, -pi, pi)
gen.add( "theta_y", double_t, 0, "twist in y", pi, -pi, pi)
gen.add( "theta_z", double_t, 0, "twist in z", 0.0, -pi, pi)

#size_enum = gen.enum([gen.const("Small",int_t,0,"A small constant"),
#          gen.const("Medium",int_t,1,"A medium constant"),
#          gen.const("Large",int_t, 2, "A large constant"),
#          gen.const("ExtraLarge",int_t,3,"An extra large constant")],
#                     "An enum to set size")
# 
#gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "dynamic_calibration", "Tutorials")) 
#first parameter is namespace's name
#second parameter is node's name
#third parameter is the current file's name
#chmod a+x ...cfg

